### ISSUE TITLE
Sometimes "using" doesn't work

### ISSUE THREAD

**Issue by Nyaacinth** [pigpigyyy/Yuescript#62]
If you write something like this
```moonscript
local a
func = (using nil) ->
    a or= 2
```
will be compiled to
```lua
local a
local func
func = function()
  a = a or 2
end
```
and the expected behaviour:
```lua
local a
local func
func = function()
  local a = a or 2
end
```

**Comment by pigpigyyy** [comment]
The problem is the update statement not creating local variables but only reusing variables instead. This will have a different and confusing behavior with Moonscript, so I change it back to the original way with commit f9859f5c6739c9a0d491ba64e241d147d4114ecb

**Comment by Nyaacinth** [comment]
Fixed with v0.7.17, closing
